<launch>

  <arg name="robot_ip" />
  
  <arg name="robot_dof" />
  <!-- load the default move_group planner (not xarm_simple_planner) -->
  <arg name="show_rviz" default="true" />
  <!-- NO_GUI_CTRL means that Rviz configuration is just for display purpose, no (Moveit) control panel loaded -->
  <arg name="no_gui_plan" default="true" />

  <arg name="add_gripper" default="false" />

  <include if="$(eval arg('robot_dof')==7 )" file="$(find xarm7_moveit_config)/launch/realMove_exec.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <!-- load the default move_group planner (not xarm_simple_planner) -->
    <arg name="show_rviz" value="$(arg show_rviz)" />
    <!-- NO_GUI_CTRL means that Rviz configuration is just for display purpose, no (Moveit) control panel loaded -->
    <arg name="no_gui_plan" value="$(arg no_gui_plan)" />
  </include>

  <include if="$(eval arg('robot_dof')==6 and arg('add_gripper')==false)" file="$(find xarm6_moveit_config)/launch/realMove_exec.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <!-- load the default move_group planner (not xarm_simple_planner) -->
    <arg name="show_rviz" value="$(arg show_rviz)" />
    <!-- NO_GUI_CTRL means that Rviz configuration is just for display purpose, no (Moveit) control panel loaded -->
    <arg name="no_gui_plan" value="$(arg no_gui_plan)" />
  </include>

  <include if="$(eval arg('robot_dof')==6 and arg('add_gripper')==true)" file="$(find xarm6_gripper_moveit_config)/launch/realMove_exec.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <!-- load the default move_group planner (not xarm_simple_planner) -->
    <arg name="show_rviz" value="$(arg show_rviz)" />
    <!-- NO_GUI_CTRL means that Rviz configuration is just for display purpose, no (Moveit) control panel loaded -->
    <arg name="no_gui_plan" value="$(arg no_gui_plan)" />
  </include>

  <include if="$(eval arg('robot_dof')==5 )" file="$(find xarm5_moveit_config)/launch/realMove_exec.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <!-- load the default move_group planner (not xarm_simple_planner) -->
    <arg name="show_rviz" value="$(arg show_rviz)" />
    <!-- NO_GUI_CTRL means that Rviz configuration is just for display purpose, no (Moveit) control panel loaded -->
    <arg name="no_gui_plan" value="$(arg no_gui_plan)" />
  </include>

  <remap from="/DOF" to="/xarm/DOF" />
  <!-- bring up the simple planner (based on move_group) that provide plan service and execution server -->
  <node name="xarm_move_group_planner" pkg="xarm_planner" type="xarm_simple_planner" />

</launch>

